### PR TITLE
Add changelog unreleased section and validation

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Adds section for running changelog in CHANGELOG.md and refactors verification scripts.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,9 +134,9 @@ commands:
           name: Install python dependencies
           command: python3 -m pip install requests
       - run:
-          name: Verify that a changelog entry is present in the PR description
+          name: Verify that a changelog entry is present in CHANGELOG.md
           command: |
-            MBX_CI_GITHUB_TOKEN=$(./mbx-ci github reader token)
+            MBX_CI_GITHUB_TOKEN=$(./mbx-ci github reader public token)
```
Comment (Zayankovsky) [inline .circleci/config.yml:139]: Is this needed for `reader` token? As far as I know this is required only for `writer` and `issues` ones.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+## Unreleased
```
Comment (Zayankovsky) [inline CHANGELOG.md:5]: Should we add changelogs from PRs that were merged recently?

File: .circleci/config.yml (hunk 1 above)
Comment (LukasPaczos) [reply .circleci/config.yml]: It's not, just fixed it while you were commenting!

File: CHANGELOG.md (hunk 2 above)
Comment (LukasPaczos) [reply CHANGELOG.md:5]: I was planning to do this separately in case there are overlaps and something lands before this one does.

**Review by Zayankovsky — APPROVED** [review]

**Pull request merged**